Vertexvis/vertex-web-sdk
# [PLAT-1779] Constrain pins to the viewer viewport

## Summary

Adjusts the behavior for text pins to constrain the label to the viewport, by creating a separate set of utilities from the markup components.

## Test Plan

- Add a pin to the right-hand side of the viewer, then resize from right to left (the pin should remain in the viewer viewport)
- Add a pin to the left-hand side of the viewer, then resize from left to right (the pin should remain in the viewer viewport)
- Attempt to move a pin label outside the viewer viewport (this should no longer be possible)

## Release Notes

N/A

## Possible Regressions

Pins/text pins

## Dependencies

N/A
